### PR TITLE
Closes #28 update advs vignette

### LINKED ISSUE #28: Documentation: Update ADVS vignette
### Background Information

Several updates are needed to the ADVS vignette. This includes:

- Use new {admiral} AVALCAT derivation function.
    - Currently there is nothing specific in {admiral} for AVALCAT derivation so ADVS template and vignette just go about it using tidyverse tools. This will change once the functionality is added to {admiral} - [see here](https://github.com/pharmaverse/admiral/issues/2480). **Update**: This is now added to Admiral dev version and can be completed.
- Use `restrict_derivation()` in conjunction with `derive_vars_crit_flag()`.
    - Without using `restrict_derivation` the crit flag is present on all PARAMCD's. See ADVS template for how to implement.
 - Update PARACAT1 to "Anthropometric measurements" for PARAMN in range 1-5. This would also close #32 

### Definition of Done

Update vignette as per specification defined above.

### PR DESCRIPTION
**Pull request by Siddhesh2097**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral family codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral<ext> (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the Reference page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: vignettes/advs.Rmd [hunk 1]
```diff
@@ -108,11 +108,11 @@ One key addition in metabolic trials are vital sign parameters associated to bod
 ```{r, echo=TRUE, message=FALSE}
 param_lookup <- tribble(
   ~VSTESTCD, ~PARAMCD, ~PARAM, ~PARAMN, ~PARCAT1, ~PARCAT1N,
-  "HEIGHT", "HEIGHT", "Height (cm)", 1, "Subject Characteristic", 1,
-  "WEIGHT", "WEIGHT", "Weight (kg)", 2, "Subject Characteristic", 1,
-  "BMI", "BMI", "Body Mass Index(kg/m^2)", 3, "Subject Characteristic", 1,
-  "HIPCIR", "HIPCIR", "Hip Circumference (cm)", 4, "Subject Characteristic", 1,
-  "WSTCIR", "WSTCIR", "Waist Circumference (cm)", 5, "Subject Characteristic", 1,
+  "HEIGHT", "HEIGHT", "Height (cm)", 1, "Anthropometric measurement", 1,
```
Comment (manciniedoardo) [inline vignettes/advs.Rmd:111]: Minor, but can we capitalise "Measurement" throughout?

**Review by manciniedoardo — CHANGES_REQUESTED** [review]
LGTM - one minor comment.

Upon reviewing, I am also thinking our test data could be improved. As it stands, everyone is in obesity class II and noone shows a 5% (let alone 10 or 15) reduction in BMI.

@Siddhesh2097 @AndersAskeland @pharmaverse/admiralmetabolic  what do you think? If weagree, could you make an issue and mark as "good first issue"?

**Comment by Siddhesh2097** [comment]
Yes, I agree. vs needs an update.

**Comment by manciniedoardo** [comment]
> Yes, I agree. vs needs an update.

could you pls make an issue? might be a good one for one of our newcomers - @kiwi5104679 maybe you could take it? I can help support.

**Comment by kiwi5104679** [comment]
@manciniedoardo Please let me do it. Once the issue is created, I will take it over.

**Comment by Siddhesh2097** [comment]
> Upon reviewing, I am also thinking our test data could be improved. As it stands, everyone is in obesity class II and noone shows a 5% (let alone 10 or 15) reduction in BMI.
- Hi @manciniedoardo , how do you want to see the reduction in BMI here ? As the CRIT flags only check for WEIGHT in current vignette and we have subjects that show reduction in weight by 5% and 10%. And that goes the same for BMI CRIT flag, we have subjects that show reduction in bmi by 5% and 10%. So is there a need to update the data ?
- @AndersAskeland

**Comment by manciniedoardo** [comment]
> > Upon reviewing, I am also thinking our test data could be improved. As it stands, everyone is in obesity class II and noone shows a 5% (let alone 10 or 15) reduction in BMI.
> 
> * Hi @manciniedoardo , how do you want to see the reduction in BMI here ? As the CRIT flags only check for WEIGHT in current vignette and we have subjects that show reduction in weight by 5% and 10%. And that goes the same for BMI CRIT flag, we have subjects that show reduction in bmi by 5% and 10%. So is there a need to update the data ?
> * @AndersAskeland

Hi @Siddhesh2097 (and @kiwi5104679 ) please see the screenshots below:

Would be good to have some in obesity classes 4 and 6 for BMI:
image

Would be good to have someone actually achieve a 5% weight reduction from baseline so that their criterion flag is Y:
image

**Comment by Siddhesh2097** [comment]
> Hi @Siddhesh2097 (and @kiwi5104679 ) please see the screenshots below:
> 
> Would be good to have some in obesity classes 4 and 6 for BMI: image
> 

-  Hi @manciniedoardo , I will create an issue now. 
But as per your query with the data, I agree it doesn't show up straight in vignette display. But in data we do have subjects in class 5 and 6 as seen below.  

image

> Would be good to have someone actually achieve a 5% weight reduction from baseline so that their criterion flag is Y: image
> 
> Does that make sense?

- And for weight reduction as well, we do not see in vignette straight away but in data we have 2 subjects who show weight reduction by 5% and one subject with 10% as well.

image

- Let me know if it will be good to update the vignette display section for categorical and criterion sections to show subjects with weight reduction and avalcat of class 5 and 6. But I think with the filter, we can easily check the data.

File: vignettes/advs.Rmd [hunk 2]
```diff
@@ -342,8 +342,8 @@ advs <- advs %>%
 ```{r, eval=TRUE, echo=FALSE}
 dataset_vignette(
   arrange(advs, USUBJID, VISITNUM, VSTPTNUM, PARAMCD),
-  display_vars = exprs(USUBJID, PARAMCD, PCHG, CRIT1, CRIT1FL, CRIT2, CRIT2FL),
-  filter = PARAMCD %in% c("WEIGHT")
+  display_vars = exprs(USUBJID, PARAMCD, PCHG, CRIT1, CRIT1FL, CRIT2, CRIT2FL, VISIT, VISITNUM),
+  filter = PARAMCD %in% c("WEIGHT") & USUBJID == "01-701-1034"
 )
```
Comment (Siddhesh2097) [inline vignettes/advs.Rmd:347]: @manciniedoardo  Added the filter for subject 1034 to display rows with CRTI1FL = "Y". If this is something that we want to show.

Comment (AndersAskeland) [reply vignettes/advs.Rmd:347]: I think that is a good idea. Perhaps also include someone without a critflag, just to show that there is a difference.

**Comment by manciniedoardo** [comment]
> > Hi @Siddhesh2097 (and @kiwi5104679 ) please see the screenshots below:
> > Would be good to have some in obesity classes 4 and 6 for BMI: image
> 
> * Hi @manciniedoardo , As per your query with the data, I agree it doesn't show up straight in vignette display. But in data we do have subjects in class 5 and 6 as seen below.
> 
> image
> 
> > Would be good to have someone actually achieve a 5% weight reduction from baseline so that their criterion flag is Y: image
> > Does that make sense?
> 
> * And for weight reduction as well, we do not see in vignette straight away but in data we have 2 subjects who show weight reduction by 5% and one subject with 10% as well.
> 
> image
> 
> * Let me know if it will be good to update the vignette display section for categorical and criterion sections to show subjects with weight reduction and avalcat of class 5 and 6. But I think with the filter, we can easily check the data.

@Siddhesh2097 Yes, I think it would be a good idea thanks.

**Review by manciniedoardo — APPROVED** [review]
LGTM! @AndersAskeland pls merge when happy 😄
